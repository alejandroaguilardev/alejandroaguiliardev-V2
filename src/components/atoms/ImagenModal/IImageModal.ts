import { IDataGalleryEducation } from "../../../data/dataGalleryEducation";

export interface IImageModal {
	open: boolean;
	handleClose: () => void;
	item: IDataGalleryEducation;
}
