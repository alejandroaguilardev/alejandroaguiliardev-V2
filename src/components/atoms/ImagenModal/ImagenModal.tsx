import { Modal, Box } from "@mui/material"
import Image from "next/image";
import { IImageModal } from "./IImageModal";


const style = {
    position: 'absolute' as 'absolute',
    top: '50%',
    left: '50%',
    transform: 'translate(-50%, -50%)',
    bgcolor: 'transparent',
    boxShadow: 24,

};

export const ImagenModal = ({ open, handleClose, item }: IImageModal) => {
    return (
        <Modal
            open={open}
            onClose={handleClose}
            aria-labelledby="parent-modal-title"
            aria-describedby="parent-modal-description"
        >
            <Box sx={{ ...style }}>
                <Image
                    src={`${item.img}`}
                    alt={item.title}
                    width={1000}
                    height={800}
                />
            </Box>
        </Modal>
    )
}
