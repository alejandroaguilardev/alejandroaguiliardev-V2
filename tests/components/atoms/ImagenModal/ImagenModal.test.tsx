import { ImagenModal } from "../../../../src/components/atoms/ImagenModal/ImagenModal";
import { render, screen } from "@testing-library/react";

describe("ImagenModal", () => {
	const item = {
		img: "/img/banner.png",
		title: "title Img",
		author: "Alejandro Aguilar",
	};

	test("Snapchoks", () => {
		const { container } = render(
			<ImagenModal open={true} handleClose={() => jest.fn()} item={item} />
		);
		expect(container).toMatchSnapshot();
	});
	
});
