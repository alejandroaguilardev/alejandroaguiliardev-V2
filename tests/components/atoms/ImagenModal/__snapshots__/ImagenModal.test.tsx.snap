// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagenModal Snapchoks 1`] = `
<div
  aria-hidden="true"
/>
`;
